app/src/main/java/swati4star/createpdf/util/PermissionsUtils.java

```diff
@@ -0,0 +1,52 @@
+package swati4star.createpdf.util;
+
+import android.app.Activity;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.content.ContextCompat;
+import android.util.Log;
+
+public class PermissionsUtils {
+    private static final String TAG = "PermissionsUtils";
+    private static Activity mActivity;
+
+    /**
+     * checkRuntimePermissions takes in multiple permissions or an array of permissions
+     * and checks for if the permission is granted and if not it requests for the desired
+     * permissions. The try and catch block checks for invalid permissions passed.
+     *
+     * @param activity
+     * @param permissions
+     * @return
+     */
+    public static boolean checkRuntimePermissions(Activity activity, String... permissions) {
+        mActivity = activity;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            for (int i = 0; i < permissions.length; i++) {
+                try {
+                    if ((ContextCompat.checkSelfPermission(activity.getApplicationContext(),
+                            permissions[i])
+                            != PackageManager.PERMISSION_GRANTED)) {
+                        requestRuntimePermissions(permissions[i]);
+                        return false;
+                    }
+                } catch (Exception e) {
+                    e.printStackTrace();
+                    Log.d(TAG, "Faulty permission passed");
+                }
+            }
+        }
+        return true;
+    }
+
+    /**
+     * requestRuntimePermissions request for the required permission
+     * if not granted.
+     *
+     * @param permission
+     */
+    public static void requestRuntimePermissions(String permission) {
+        ActivityCompat.requestPermissions(mActivity, new String[]{permission}, 0);
```

Comment:
@sidhuparas It may happen that a particular permission out of a group of permissions is required. Then why to request for a whole group of permissions. Here the `checkPermissions` function takes care of all by requesting asked desired permissions iteratively each desired one only at a time.

Reply:
@sidhuparas Moreover how to detect for a `permission_group` if a particular permission is passed (eg: - xyz(a valid permission) is passed)

Reply:
If you ask for permissions iteratively, it would ask for one permission and then another one by one. A more intuitive method is combined permissions: https://stackoverflow.com/a/35495855